src/main/java/net/tropicraft/core/common/biome/decorators/BiomeDecoratorRainforest.java

```diff
@@ -42,39 +41,38 @@ public void genDecorations(Biome biome, World world, Random rand) {
 		}
 
 		if(rand.nextInt(HOME_TREE_RARITY) == 0) {
-			int cx = x/* * 16*/;
-			//	System.out.println("Trying!!!");
-			int cz = z/* * 16*/;
+			int cx = x;
+			int cz = z;
 			int xx = rand.nextInt(16) + cx + 8;
-			int zz= rand.nextInt(16) + cz + 8;
+			int zz = rand.nextInt(16) + cz + 8;
 			new WorldGenHomeTree(world, rand).generate(new BlockPos(xx, 0, zz));
 		}
 		//
 		//		if(rand.nextInt(ALTAR_CHANCE) == 0) {
 		//			new WorldGenForestAltarRuin(world, rand).generate(randDecorationCoord(rand, x, 16), 0, randDecorationCoord(rand, x, 16));
 		//		}
 		//
-		if (rand.nextInt(TALL_TREE_CHANCE) == 0) {
-			i = randDecorationCoord(rand, x, 16);
-			k = randDecorationCoord(rand, z, 16);
-			new WorldGenTallTree(world, rand).generate(new BlockPos(i, getTerrainHeightAt(world, i, k), k));
+		for (int l = 0; l < TropicsConfigs.rainforestThicknessAmount; l++) {
+		    i = randDecorationCoord(rand, x, 2) + 8;
+		    k = randDecorationCoord(rand, z, 2) + 8;
+		    new WorldGenTallTree(world, rand).generate(new BlockPos(i, getTerrainHeightAt(world, i, k), k));
 		}
 
-		if (rand.nextInt(UP_TREE_CHANCE) == 0) {
-			i = randDecorationCoord(rand, x, 16);
-			k = randDecorationCoord(rand, z, 16);
-			new WorldGenUpTree(world, rand).generate(new BlockPos(i, getTerrainHeightAt(world, i, k), k));			
+		for (int l = 0; l < TropicsConfigs.rainforestThicknessAmount; l++) {
+		    i = randDecorationCoord(rand, x, 4) + 8;
+		    k = randDecorationCoord(rand, z, 4) + 8;
+		    new WorldGenUpTree(world, rand).generate(new BlockPos(i, getTerrainHeightAt(world, i, k), k));
 		}
 
-		for (int a = 0; a < SMALL_TUALANG_AMOUNT; a++) {
-			i = randDecorationCoord(rand, x, 16);
-			k = randDecorationCoord(rand, z, 16);
+		for (int a = 0; a < SMALL_TUALANG_AMOUNT * TropicsConfigs.rainforestThicknessAmount * 2; a++) {
+			i = randDecorationCoord(rand, x, 4) + 6;
+			k = randDecorationCoord(rand, z, 4) + 6;
 			new WorldGenTualang(world, rand, 16, 9).generate(new BlockPos(i, getTerrainHeightAt(world, i, k), k));
 		}
 
-		for (int a = 0; a < LARGE_TUALANG_AMOUNT; a++) {
-			i = randDecorationCoord(rand, x, 16);
-			k = randDecorationCoord(rand, z, 16);
+		for (int a = 0; a < LARGE_TUALANG_AMOUNT * TropicsConfigs.rainforestThicknessAmount * 2; a++) {
+            i = randDecorationCoord(rand, x, 4) + 6;
```

Comment:
Why +6?

Reply:
Wanted something that might not be exactly in the middle of the chunk each time. Guess +8 is fine too